src/Pim/Bundle/CatalogBundle/Elasticsearch/Sorter/IdentifierSorter.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\Elasticsearch\Sorter;
+
+use Pim\Bundle\CatalogBundle\Elasticsearch\SearchQueryBuilder;
+use Pim\Component\Catalog\Exception\InvalidDirectionException;
+use Pim\Component\Catalog\Model\AttributeInterface;
+use Pim\Component\Catalog\Query\Sorter\AttributeSorterInterface;
+use Pim\Component\Catalog\Query\Sorter\Directions;
+use Pim\Component\Catalog\Query\Sorter\FieldSorterInterface;
+
+/**
+ * Identifier sorter for an Elasticsearch query.
+ *
+ * As the identifier product value is not indexed in ES. Whenever you want to sort on the attribute identifier 'sku'
+ * or on the field 'Identifier', the PQB will use this class which sorts on the field identifier.
+ *
+ * @author    Samir Boulil <samir.boulil@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class IdentifierSorter implements FieldSorterInterface, AttributeSorterInterface
+{
+    /** @var SearchQueryBuilder */
+    protected $searchQueryBuilder;
+
+    /** @var array */
+    protected $supportedFields;
+
+    /** @var array */
+    protected $supportedAttributes;
+
+    /**
+     * @param array $supportedFields
+     * @param array $supportedAttributes
+     */
+    public function __construct(array $supportedFields = [], array $supportedAttributes = [])
+    {
+        $this->supportedFields = $supportedFields;
+        $this->supportedAttributes = $supportedAttributes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setQueryBuilder($searchQueryBuilder)
+    {
+        if (!$searchQueryBuilder instanceof SearchQueryBuilder) {
+            throw new \InvalidArgumentException(
+                sprintf('Query builder should be an instance of "%s"', SearchQueryBuilder::class)
+            );
+        }
+
+        $this->searchQueryBuilder = $searchQueryBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addFieldSorter($field, $direction, $locale = null, $channel = null)
```

Comment:
This looks to me identical to the BaseFieldSorter. Could you not just extend it and add `implements AttributeSorterInterface` + the two needed methods?